### ISSUE TITLE
Blocked a frame with origin from accessing a cross-origin frame

### ISSUE THREAD

**Issue by NicolasGoerlach** [codefog/contao-news_categories#100]
Hi, bekomme aktuell auf meiner Contao Installation (3.5.24) folgenden Fehler in der Console wenn ich einer Nachricht Kategorien zuweisen möchte und anwenden klicke:

> Uncaught DOMException: Blocked a frame with origin "http://192.168.50.11" from accessing a cross-origin frame. at Object.<anonymous>(http://192.168.50.11/system/modules/widget_tree_picker/assets/treepicker.min.js:1:2040) at HTMLAnchorElement.a (http://192.168.50.11/assets/js/6285e9ffc6ee.js:3:27847)

Hast du eine Idee woran das liegen könnte?

**Comment by NicolasGoerlach** [comment]
Hm.. Sehe gerade, dass ich diesen Fehler in jeder Backend-Lightbox habe. Hat also wohl nichts mit der Erweiterung zutun, kannst closen.

**Comment by qzminski** [comment]
Do other modal windows work for you normally, say contao page/file picker?

**Comment by fritzmg** [comment]
The problem is actually caused when the [fritzmg/contao-facebook-js-sdk](https://github.com/fritzmg/contao-facebook-js-sdk) Extension and the [codefog/contao-widget_tree_picker](https://github.com/codefog/contao-widget_tree_picker) is installed together.

The former integrates the Facebook JavaScript SDK in the front end as well as the back end. When the Facebook JS SDK is loaded, it creates hidden iframes over which the SDK communicates with Facebook. See [stackoverflow.com/a/16865639/374996](https://stackoverflow.com/a/16865639/374996).

The `widget_tree_picker` extension searches for the `simple-modal-iframe` here: [/assets/treepicker.js#L132-L138](https://github.com/codefog/contao-widget_tree_picker/blob/2.3.2/assets/treepicker.js#L132-L138). Since it iterates over _all_ present iframes and tries to access their name, this leads to the aforementioned CORS violation.

**Comment by fritzmg** [comment]
> Thanks @fritzmg for valuable input! Any ideas how to solve it?

I have no good idea yet. Imho, it is not advisable to simply iterate over all available iframes, since there is a chance one of these iframes is not from the same origin. However, I am guessing Contao forces you to do so, since it does not provide an interface to get the `simple-modal-iframe`. May be this is something that has to be implemented in the Contao Core first (i.e. in the `contao/core-bundle`).

However, I am not entirely sure yet if this is the actual problem - or if accessing the regular Contao iframes already causes the CORS violation, since the Facebook JS SDK sets `document.domain = 'facebook.com'` - I don't understand the details of this method yet (see also [stackoverflow.com/a/3076648/374996](http://stackoverflow.com/a/3076648/374996)).

> What is the reason you inject Facebook JS SDK in the backend?

In short: it's for an extension that lets you post news to Facebook Pages automatically. In order to be able to do that, you need to fetch a long term access token with your Facebook login. The login and the fetching of the appropriate long term access token is done via the Facebook JavaScript API directly in the back end of Contao (in the settings of the news archive).

May be I can do it without the JavaScript API, but this requires some major changes.

**Comment by fritzmg** [comment]
I had a brief conversation with the Facebook engineer [here](http://stackoverflow.com/questions/16862207/facebook-and-cross-domain-messaging-clarification/16865639?noredirect=1#comment73113539_16865639) and we agreed that it is generally not safe to assume that all present iframes belong to you.

When I have time, I'll try to implement something in the `contao/core-bundle` which could prevent having to iterate over all present iframes in JavaScript.

**Comment by NicolasGoerlach** [comment]
Wow, didn't thought that the problem is so deep :D Sorry guys! 
And many thanks to @fritzmg .

**Comment by qzminski** [comment]
I am closing this as we will use the new Contao DCA picker in Contao 4.4.
